### PR TITLE
Queue | Move VACOLS issue descriptions, remand reasons into shared constant

### LINKED ISSUE #5194: Move VACOLS issue descriptions into shared constant
Currently we maintain a similar list of issue descriptions on [the front-end](https://github.com/department-of-veterans-affairs/caseflow/blob/42f274d1cc27c99eaa45b6ca633048842fec07b3/client/app/queue/constants.js#L125) and [the back-end](https://github.com/department-of-veterans-affairs/caseflow/blob/42f274d1cc27c99eaa45b6ca633048842fec07b3/lib/constants/issue.rb#L3). The work on this ticket entails:
1. Determining the structure this data should take
2. Resolving the differences between the values where they exist (e.g. 11. "NCA Burial" vs "Burial benefits")
3. Using [the shared constants pattern](https://github.com/department-of-veterans-affairs/caseflow/pull/5127) to allow the front- and back-end of the application to use the same set of values from a single, shared file (parsing in js/ruby as required)

## Acceptance Criteria
* Front- and back-end load this set of values from a single, shared file

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #5194

### Description
Moves `ISSUE_INFO` and `REMAND_REASONS` into the new shared constants format

### Acceptance Criteria 
- [ ] Add/Edit Issue View and Remand Reasons screen behaves as expected

### Testing Plan
1. Go to `/queue`, begin Draft Decision checkout flow
2. Edit issue, confirm dropdown options appear as before
3. Select "Remanded" as disposition, go to Remand Reasons screen, confirm reasons appear as expected

### CODE REVIEW

File: constants/ISSUE_INFO.json [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+        "levels": {
+          "01": { "description": "Service connected" },
+          "02": { "description": "Nonservice connected" }
+        }
+      },
+      "02": {
+        "description": "Other"
+      }
+    }
+  },
```
Comment (lowellrex) [inline constants/ISSUE_INFO.json]: It seems to me like this constants is a series of one single data structure nested inside itself a bunch. For the sake of example, I'll call that `Issue`. I think that looks something like:
```
Issue {
  description: String,
  (optional) levels: Map { key: Issue, ... }
}
```
The only place this model deviates is in the top level issue where what we call "levels" is instead called "issues". Can we settle on one name for that map of Issues (either levels or issues, I don't have a preference).

I think a uniform model makes parsing this much easier (especially on [the backend](https://github.com/department-of-veterans-affairs/caseflow/blob/42f274d1cc27c99eaa45b6ca633048842fec07b3/lib/constants/issue.rb#L3) where we only care about the description).

File: constants/REMAND_REASONS_BY_ID.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (lowellrex) [inline constants/REMAND_REASONS_BY_ID.json]: Do we use these values anywhere on the backend?

File: client/app/queue/constants.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 /* eslint-disable max-lines */
 import { css } from 'glamor';
 import VACOLS_DISPOSITIONS_BY_ID from '../../../constants/VACOLS_DISPOSITIONS_BY_ID.json';
+import REMAND_REASONS_BY_ID from '../../../constants/REMAND_REASONS_BY_ID.json';
+// https://git.io/vxBtN
+export ISSUE_INFO from '../../../constants/ISSUE_INFO.json';
```
Comment (lowellrex) [inline client/app/queue/constants.js]: What advantages are there when importing from json and exporting as an object in this file as opposed to importing from json in the files that use the values (like we do in [IssueListItem](https://github.com/department-of-veterans-affairs/caseflow/blob/c0e0bfd8c7a8a0c33ea2eba6cb28f9f0f5ffcb3f/client/app/queue/components/IssueListItem.jsx#L12) and [SelectIssueDispositionDropdown](https://github.com/department-of-veterans-affairs/caseflow/blob/c0e0bfd8c7a8a0c33ea2eba6cb28f9f0f5ffcb3f/client/app/queue/components/SelectIssueDispositionDropdown.jsx#L10))?

File: constants/REMAND_REASONS_BY_ID.json (hunk 2 above)
Comment (dannysellers) [reply constants/REMAND_REASONS_BY_ID.json]: We just use the codes for validation, within [`remand_reason.rb`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/vacols/remand_reason.rb#L4). That list includes 'inactive' reasons that aren't used / aren't in the list we had for the front end. I'll add inactive reasons so we've got them all in one place.

**Comment by lowellrex** [comment]
Made a few comments above but I like it! I'd like to see this PR include [changes to the backend](https://github.com/department-of-veterans-affairs/caseflow/blob/42f274d1cc27c99eaa45b6ca633048842fec07b3/lib/constants/issue.rb#L3) to use the ISSUE_INFO constant there.

The big benefit of changing both back- and front-end at the same time is that we don't commit to a data structure that isn't super flexible for one of the two, and don't bias too heavily toward the existing structure because a slight change to the structure to satisfy both front- and back-end would require a change to the existing implementation.

File: constants/REMAND_REASONS_BY_ID.json (hunk 2 above)
Comment (lowellrex) [reply constants/REMAND_REASONS_BY_ID.json]: Rad, and then we could replace that array of codes with the keys from the constants file! I'm irrationally pumped about this.

File: client/app/queue/constants.js (hunk 3 above)
Comment (dannysellers) [reply client/app/queue/constants.js]: Initially I was thinking it'd be more straightforward to have external/shared constants imported through an 'app'-specific constants file (i.e. `/queue/constants`), though I'm not sure that's necessary at this point

**Comment by dannysellers** [comment]
I agree that it'd be nice to consolidate [`ISSUE_INFO`](https://github.com/department-of-veterans-affairs/caseflow/blob/e5e205d77a847c886d29c77d6173b5edfafac91d/constants/ISSUE_INFO.json) with [`Constants::Issue::ISSUE_DESCRIPTIONS`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/lib/constants/issue.rb), though I'm not really sure what the relation is between the two datasets / why they're different. They don't quite match up in structure or content 😕, though it seems like they should

**Comment by lowellrex** [comment]
Can we add an optional "plain_language_description" element (or maybe "veteran_friendly_description"?) to each Issue in the set of ISSUE_INFO values? We could grab these values from the [back-end ISSUE_DESCRIPTIONS](https://github.com/department-of-veterans-affairs/caseflow/blob/master/lib/constants/issue.rb) when the description there differs from what we have in the constants file now.

So the Issue data structure would look like:
```
Issue {
  description: String,
  (optional) plain_language_description: String,
  (optional) levels: Map { key: Issue, ... }
}
```

I think the only thing we will have to change on the back-end is `Issue.friendly_description_for_codes()` where we could do something like the following:
```
def friendly_description_for_codes(code_array)
  issue_description = code_array.reduce(Constants::ISSUE_INFO) do |levels, code|
    child_levels = levels[code]['levels']
    unless child_levels
      description = levels[code]['plain_language_description'] || levels[code]['description']
      break description if description.is_a?(String)
      return nil
    end
    child_levels
  end

  if diagnostic_code
  ...
```

What do you think?

**Pull request merged**